### ISSUE TITLE
SRU and OAI interface is not layer aware

### ISSUE THREAD

**Issue by phochste** [LibreCat/LibreCat#278]
The wiki suggest that the SRU and OAI interfaces are layer aware : https://github.com/LibreCat/LibreCat/wiki/Add-Change--OAI--SRU

But, this is not the case. The config.yml file can be moved to a layer and edited but all the fixes and templates used in this configuration file are ignorant of the layers (and other layer configuration files).

E.g. all the `dc:idenifiers` start with:
 
    http://www.librecat.org/

by the `host` configuration option in the `catmandu.yml` file. Overwriting this option in a layer doesn't update the `dc:identifiers`.

**Comment by nicolasfranck** [comment]
cf. https://github.com/LibreCat/LibreCat/pull/291

That string "http://www.librecat.org" is actually hardcoded into the fixes..

Should be fixed after the merge of that pull request

**Comment by vpeil** [comment]
Now problem now is that no host ist shown at all, just
```
<dc:identifier>publication/2737386</dc:identifier>
```

**Comment by nicolasfranck** [comment]
Using request information in OAI is a bit painfull.
One has to override the fixes in the environment Dancer file:
```
environments/development.yml
```
to something like:
```
plugins:
  "Catmandu::OAI":
    metadata_formats:
      -
        metadataPrefix: epicur
        schema: http://www.persistent-identifier.de/xepicur/version1.0/xepicur.xsd
        metadataNamespace: urn:nbn:de:1111-2004033116
        template: views/export/oai_epicur.tt
        cql: 'status exact public AND fulltext exact 1'
        fix: ["add_field('uri_base','http://localhost:5001')"]
```
But because merging of configuration does not merge arrays,
you would have to copy the WHOLE list of metadata_formats

Sounds like a job for the Catmandu::OAI plugin: add environment variables
to every hit (or make it optional at least).

**Comment by phochste** [comment]
I don't mind that some things are hard coded in the templates. The fact is that the templates can't be put in a layer. It always uses the templates of the main code. Also configuration options can't be put in a layer

**Comment by nics** [comment]
Dancer::Plugin::Catmandu::OAI 0.05 has a new template_options setting that allows you to passe the include path:

    template_options:
        INCLUDE_PATH: ['layer1/views', 'layer2/views']

**Comment by nics** [comment]
Dancer::Plugin::Catmandu::SRU 0.05 has the same template_options setting

**Comment by vpeil** [comment]
> The problem now is that no host ist shown at all, just
<dc:identifier>publication/2737386</dc:identifier>

It's correct now on dev (but since when?)

**Comment by vpeil** [comment]
@phochste is that issue fixed for you? The issue description mentions `host` which is layer-aware. Is there more?

**Comment by phochste** [comment]
I put this in my template options:

```(yaml)
   template_options:
        INCLUDE_PATH:
            - /opt/librecat-test
```

And change the view/exports/oai_dc.tt templates in the layer. Executed:

http://localhost:5001/oai?verb=ListRecords&metadataPrefix=oai_dc

and still get the result from the templates in the main layer. (If upgraded all the plugins to the latest version)

**Comment by vpeil** [comment]
I think, you should use

```
template_options:
    INCLUDE_PATH:
        - /opt/librecat-test/views
```

**Comment by phochste** [comment]
Nope still get the main layer templates...

**Comment by phochste** [comment]
I checked the code. The correct INCLUDE_PATH-s are being provided at this spot: https://github.com/LibreCat/Dancer-Plugin-Catmandu-OAI/blob/master/lib/Dancer/Plugin/Catmandu/OAI.pm#L321
But, they have no effect the Dancer application keeps using the old settings for INCLUDE_PATH-s

**Comment by phochste** [comment]
Will be fixed with Dancer::Plugin::Catmandu::OAI 0.0501
